src/core/lib/surface/call.cc

```diff
@@ -99,8 +99,14 @@ struct batch_control {
   } completion_data;
   grpc_closure start_batch;
   grpc_closure finish_batch;
-  gpr_refcount steps_to_complete;
+  grpc_core::Atomic<uintptr_t> steps_to_complete;
```

Comment:
Why not use the RefCount class in src/core/lib/gprpp/ref_counted.h?https://github.com/grpc/grpc/blob/f4ce5fc2510c342adc86afa41a9354d9956a9092/src/core/lib/gprpp/ref_counted.h#L73

Reply:
I think the problem is that we decided not to allow Ref overrides, but here we need to be able to reset the steps in line 1875.

But, I'd suggest using `intptr_t` instead of `uintptr_t`.

Reply:
Could you elaborate more? I'm not quite understanding.

Reply:
Changed to use intptr_t as requested.

@hcaseyal steps_to_complete gets repeatedly set to non-zero values, then decrements to 0, but unlike grpc_core::RefCount it can be set and used again. I believe for grpc_core::RefCount, once it goes to 0, it's done. Hence not using it.